src/transaction/log_manager.c

```diff
@@ -10214,6 +10217,129 @@ logtb_tran_update_stats_online_index_rb (THREAD_ENTRY * thread_p, void *data, vo
   return error_code;
 }
 
+#if defined (SERVER_MODE)
+static void
+cdc_thread_checker (THREAD_ENTRY * thread_p)
+{
+  if (LSA_LE (&log_Reader_info.next_lsa, log_Gl.append.get_nxio_lsa ()))
+    {
+      //pthread_cond_signal (&lsa_cond);
+    }
+}
+
+static void
+cdc_log_producer (THREAD_ENTRY * thread_p)
+{
+
+}
+
+/* *INDENT-OFF* */
+static void
+cdc_log_producer_execute (cubthread::entry & thread_ref)
+{
+  if (!BO_IS_SERVER_RESTARTED ())
+  {
+    return;
+  }
+  
+  cdc_log_producer (&thread_ref);
+}
+
+static void
+cdc_thread_checker_execute (cubthread::entry & thread_ref)
+{
+  if (!BO_IS_SERVER_RESTARTED ())
+  {
+    return;
+  }
+  
+  cdc_thread_checker (&thread_ref);
+}
+/* *INDENT-ON* */
+
+/* *INDENT-OFF* */
+void
+cdc_log_producer_interval (bool & is_timed_wait, cubthread::delta_time & period)
+{
+
+  int cdc_log_producer_interval_sec = 0; /* for future fix to decrease server load */
+
+  if (cdc_log_producer_interval_sec > 0)
```

Comment:
This is an unused code.

The 'log_producer' thread wakes up periodically and can check the condition of 'next_lsa' <= 'nxio_lsa'. Is it right?
If it is, then I think you can remove the 'thread_checker' thread.